core/vm/contracts.go

```diff
@@ -1135,3 +1140,28 @@ func kZGToVersionedHash(kzg kzg4844.Commitment) common.Hash {
 
 	return h
 }
+
+// BeaconRoot is a stateful precompile that returns a beacon root.
+type beaconRoot struct{}
+
+func (c *beaconRoot) RequiredGas(input []byte) uint64 {
+	return params.BeaconRootPrecompileGas
+}
+
+func (c *beaconRoot) Run(stateDB StateReader, input []byte) ([]byte, error) {
+	if len(input) != common.HashLength {
+		return nil, errors.New("invalid input length")
+	}
```

Comment:
This is (subtly wrong). From the EIP:

> If the input is more than 32 bytes, the precompile only takes the first 32 bytes of the input buffer and ignores the rest. If the input is less than 32 bytes, the precompile should revert.

cc @shemnon will this make the next testnet go boom? Or does besu contain the same error?

Reply:
we can also update the EIP here, if that's simpler

I don't know if there is precedent for precompile input handling one way or the other...

Reply:
Besu hasn't implemented it yet.  @pinges is supposed to be implementing that.